api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -425,16 +425,22 @@ type FelixConfigurationSpec struct {
 	// RouteTableRange specifies the indices of the route tables that Calico should use.
 	RouteTableRange *RouteTableRange `json:"routeTableRange,omitempty" validate:"omitempty"`
 
-	// WireguardEnabled controls whether Wireguard is enabled. [Default: false]
+	// WireguardEnabled controls whether Wireguard is enabled for IPv4. [Default: false]
 	WireguardEnabled *bool `json:"wireguardEnabled,omitempty"`
+	// WireguardEnabledV6 controls whether Wireguard is enabled for IPv6. [Default: false]
+	WireguardEnabledV6 *bool `json:"wireguardEnabledV6,omitempty"`
 	// WireguardListeningPort controls the listening port used by Wireguard. [Default: 51820]
 	WireguardListeningPort *int `json:"wireguardListeningPort,omitempty" validate:"omitempty,gt=0,lte=65535"`
 	// WireguardRoutingRulePriority controls the priority value to use for the Wireguard routing rule. [Default: 99]
 	WireguardRoutingRulePriority *int `json:"wireguardRoutingRulePriority,omitempty" validate:"omitempty,gt=0,lt=32766"`
-	// WireguardInterfaceName specifies the name to use for the Wireguard interface. [Default: wg.calico]
+	// WireguardInterfaceName specifies the name to use for the IPv4 Wireguard interface. [Default: wireguard.cali]
 	WireguardInterfaceName string `json:"wireguardInterfaceName,omitempty" validate:"omitempty,interface"`
-	// WireguardMTU controls the MTU on the Wireguard interface. See Configuring MTU [Default: 1420]
+	// WireguardInterfaceNameV6 specifies the name to use for the IPv6 Wireguard interface. [Default: wg-v6.cali]
```

Comment:
Initially I did try it, but this had issues with the "iface-param" validation, since it cuts off at 15 characters:
https://github.com/projectcalico/calico/blob/e713b13d98844520b19351fc86ed93b771d9f796/felix/config/config_params.go#L51

I didn't feel secure changing the validation, so I ended up renaming the v6 intf. Let me know if we should go the other way on this.

Reply:
Ah gotcha, fine to leave this then.